### PR TITLE
[Merged by Bors] - refactor(CategoryTheory): more general form of the characterisation of condensed sets as sheaves on `Stonean`

### PR DESCRIPTION
**Pull request by dagurtomas**
Given a fully faithful functor `F : C ⥤ D` into a precoherent category, which preserves and reflects finite effective epi families, and satisfies the property `F.EffectivelyEnough` (meaning that to every object in `C` there is an effective epi from an object in the image of `F`), the categories of coherent sheaves on `C` and `D` are equivalent.

We give the corresonding result for the regular topology as well.

We use this general setup to refactor the characterisation of condensed sets as sheaves on `Stonean`; indeed, the inclusion functor to `CompHaus` satisfies `Functor.EffectivelyEnough` because the Stonean spaces are the projective objects in `CompHaus`, which has enough projectives and every epimorphism in `CompHaus` is effective.

---

This PR moves the two instances of `F.IsCoverDense` from the file `CategoryTheory/Sites/Coherent/CoverDense` to the new file `CategoryTheory/Sites/Coherent/SheafComparison` and removes the former file. That file structure didn't make sense anymore after I got rid of all the projectivity assumptions.

- [x] depends on: #11476 
- [x] depends on: #11491
- [x] depends on: #11502
- [x] depends on: #11567
- [x] depends on: #11568   
- [x] depends on: #11686 

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/CategoryTheory/Sites/Coherent/SheafComparison.lean [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+  [F.Full] [F.Faithful]
+  [Precoherent D]
+  [F.EffectivelyEnough]
+
+/--
+The equivalence from coherent sheaves on `Stonean` to coherent sheaves on `CompHaus`
+(i.e. condensed sets).
+-/
+noncomputable
+def equivalence (A : Type _) [Category.{u+1} A] [HasLimits A] : haveI := F.reflects_precoherent
```
Comment (riccardobrasca) [inline Mathlib/CategoryTheory/Sites/Coherent/SheafComparison.lean:109]: Can you mention the three equivalence in the doc at beginning, as main applications?

File: Mathlib/CategoryTheory/Sites/Coherent/SheafComparison.lean [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+universe u
+
+variable {C D : Type (u+1)} [LargeCategory C] [LargeCategory D] (F : C ⥤ D)
+  [F.PreservesEffectiveEpis] [F.ReflectsEffectiveEpis]
+  [F.Full] [F.Faithful]
+  [Preregular D]
+  [F.EffectivelyEnough]
+
+/--
+The equivalence from coherent sheaves on `Stonean` to coherent sheaves on `CompHaus`
```
Comment (riccardobrasca) [inline Mathlib/CategoryTheory/Sites/Coherent/SheafComparison.lean:210]: This docstring seems wrong.

File: Mathlib/CategoryTheory/Sites/Coherent/SheafComparison.lean [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+/-!
+
+# Categories of coherent sheaves
+
+Given a fully faithful functor `F : C ⥤ D` into a precoherent category, which preserves and reflects
+finite effective epi families, and satisfies the property `F.EffectivelyEnough` (meaning that to
+every object in `C` there is an effective epi from an object in the image of `F`), the categories
+of coherent sheaves on `C` and `D` are equivalent.
+
+We give the corresonding result for the regular topology as well.
```
Comment (riccardobrasca) [inline Mathlib/CategoryTheory/Sites/Coherent/SheafComparison.lean:19]: Can you add a reference to the actual declaration?

**Review by riccardobrasca — APPROVED** [review]
This is amazing, thanks!!

bors d+

**Comment by dagurtomas** [comment]
Thanks for the reviews! 

bors merge

**Pull request closed without merging**